internal/master/service.go

```diff
@@ -515,6 +554,65 @@ func (ds *distributedService) GetStatus(context context.Context, request *emptyp
 	return regionInfo, nil
 }
 
+
+func (ds *distributedService) Check(ctx context.Context, healthCheckReq *serverpb.HealthCheckRequest) (*serverpb.HealthCheckResponse, error) {
+	regionInfo := ds.DKVService.(*standaloneService).regionInfo
+
+	if ds.isClosed {
+		return &serverpb.HealthCheckResponse{Status: serverpb.HealthCheckResponse_NOT_SERVING}, nil
+	} else {
+		// Currently there is no way for this instance of DKVServer to know if its the local dc follower, or is a follower with a lot of lag
+		// Even the member list api ListMembers() is just an in memory lookup rather than actually looking at the current raft member state
+		// If the current node is itself disconnected, it will provide stale / incorrect member list
+		// For now, we will return status based on listMembers() and identify based on stale data
+		// As of now, there is no hard requirement to identify true leader via service discovery thus using listmembers() is fine
+		// TODO - Provide correct status wrt master / local dc follower / follower with lot of lag
```

Comment:
Maybe correct this comments. As we discussed this inconsistency is only for a very short interval, and shouldn't affect in case of healthcheck.

Reply:
Corrected. The client will get SERVING if the node is a follower or a leader.